lib/registry.js

```diff
@@ -361,7 +356,7 @@ Registry.prototype.createDataSource = function(name, options) {
   };
 
   if (ds.settings && ds.settings.defaultForType) {
-    this.setDefaultDataSourceForType(ds.settings.defaultForType, ds);
+    console.warn('DataSource option "defaultForType" is no longer supported');
```

Comment:
I think we should `assert` or throw... How would the app function with this setting?